src/include/gurt/telemetry_producer.h

```diff
@@ -10,6 +10,7 @@
 
 /* Developer facing server API to write data */
 void d_tm_set_counter(struct d_tm_node_t *metric, uint64_t value);
+void d_tm_get_counter_srv(struct d_tm_node_t *metric, uint64_t *value);
```

Comment:
No need to add a duplicate function. You can include telemetry_consumer.h and use `d_tm_get_counter` as needed.

Reply:
d_tm_get_counter() need a parameter client context "struct d_tm_context *ctx", that is unavailable (and useless) for daos server.

Reply:
You can open a new d_tm_context for reading with `d_tm_open(dss_instance_idx)` and close it with `d_tm_close`. If you do it during module init/fini time it shouldn't hurt performance. I'm just reluctant to create a set of duplicate read APIs, it seems destined to create confusion.

Reply:
This happens during DAOS engine running for every DTX batched committed RPC handling. It is on the hot path. So the performance has to be considered. I am not sure how much open/close the d_tm_context will affect the performance, but I am afraid to do that frequently will cause some overhead.

Reply:
Yeah, I don't think you should open/close ctx on the hot path. But is there a reason it can't be done during dtx module init/fini, and stored?

Reply:
An alternative is to update the telemetry client API to accept NULL for the ctx, and use the global server ctx in that case. As I think about it, I'm starting to think that's preferable.

Reply:
If just consider implementation, that is not impossible. But from the efficiency perspective, it is obvious some redundant than the d_tm_get_counter_srv(). Since we already get the metric handle for DTX use case, it is unnecessary to verify the node pointer and then locate the handle again via "context" and node pointer. So for me, d_tm_get_counter_srv() is a fast version for server side use case. If we need to unify related APIs, then I would suggest to extend d_tm_get_counter() with introducing new parameter "metric": if it is non-NULL, then directly use it without validate_node_ptr() and conv_ptr().

Reply:
I'd still really prefer a unified API. I agree, it is fine to skip the conversion altogether on the server side. Maybe we can combine both ideas, if the ctx == NULL, skip conversion and get the value directly.

If you don't want to implement this and update unit tests, we can land as-is, but we should unify ASAP.